apidef/migration.go

```diff
@@ -207,25 +207,20 @@ func (a *APIDefinition) migrateEndpointMetaByType(typ int) {
 }
 
 func (a *APIDefinition) Migrate() (versions []APIDefinition, err error) {
+	a.MigrateAuthentication()
+
 	versions, err = a.MigrateVersioning()
 	if err != nil {
 		return nil, err
 	}
 
-	a.MigrateEndpointMeta()
-	a.MigrateCachePlugin()
-
-	for k, v := range a.AuthConfigs {
-		v.Name = k
-		a.AuthConfigs[k] = v
-	}
-
+	versions = append(versions, *a)
```

Comment:
`*a` is likely incorrect - if you want a copy here, it will only copy internal pointers, rather the underlying data.

Reply:
Not copying, the versions slice is value slice, so it is for converting